packages/utils/src/eslint-utils/rule-tester/RuleTester.ts

```diff
@@ -1,28 +1,42 @@
 import type * as TSESLintParserType from '@typescript-eslint/parser';
+import assert from 'assert';
 import { version as eslintVersion } from 'eslint/package.json';
 import * as path from 'path';
 import * as semver from 'semver';
 
-import * as TSESLint from '../../ts-eslint';
+import type { ParserOptions } from '../../ts-eslint/ParserOptions';
+import type { RuleModule } from '../../ts-eslint/Rule';
+import type { RuleTesterTestFrameworkFunction } from '../../ts-eslint/RuleTester';
+import * as BaseRuleTester from '../../ts-eslint/RuleTester';
 import { deepMerge } from '../deepMerge';
 import type { DependencyConstraint } from './dependencyConstraints';
 import { satisfiesAllDependencyConstraints } from './dependencyConstraints';
 
 const TS_ESLINT_PARSER = '@typescript-eslint/parser';
 const ERROR_MESSAGE = `Do not set the parser at the test level unless you want to use a parser other than ${TS_ESLINT_PARSER}`;
 
-type RuleTesterConfig = Omit<TSESLint.RuleTesterConfig, 'parser'> & {
+type RuleTesterConfig = Omit<BaseRuleTester.RuleTesterConfig, 'parser'> & {
   parser: typeof TS_ESLINT_PARSER;
+  /**
+   * Constraints that must pass in the current environment for any tests to run
+   */
+  dependencyConstraints?: DependencyConstraint;
```

Comment:
We're now repeating this comment & property three times. How about extracting it into an interface named something like `WithDependencyConstraints`?

Reply:
This comment is slightly different in that it uses `any tests` instead of `the test`.

I did want to avoid *another* interface in this file cos there's a lot of stuff here.
but happy to do it if you feel strongly!

Reply:
No strong feelings 😄